### PR TITLE
Update AppNexus direct to run in UK/ROW

### PR DESCRIPTION
**Pull request by janua**
## What does this change?

This changes the adapter `and` (AppNexus Direct) to run in the UK and ROW regions with specific placement IDs.

@guardian/commercial-dev

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/prebid/bid-config.js [hunk 1]
```diff
@@ -217,7 +217,38 @@ const getImprovePlacementId = (sizes: PrebidSize[]): number => {
...
+const getAppNexusDirectPlacementId = (sizes: PrebidSize[]): string => {
+    if (isInAuRegion()) {
+        return '11016434';
+    }
+
+    const defaultPlacementId: string = '9251752';
+    switch (getBreakpointKey()) {
+        case 'D':
+            if (containsMpuOrDmpu(sizes)) {
+                return '9251752';
```
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/prebid/bid-config.js]: `return defaultPlacementId;` ???

File: static/src/javascripts/projects/commercial/modules/prebid/utils.js [hunk 2]
```diff
@@ -69,7 +69,8 @@ export const getRandomIntInclusive = (
     return Math.floor(Math.random() * (max - min + 1)) + min;
 };
 
-export const shouldIncludeAppNexus = (): boolean => isInAuRegion();
+export const shouldIncludeAppNexus = (): boolean =>
+    isInAuRegion() || !isInUsRegion();
```
Comment (katebee) [inline static/src/javascripts/projects/commercial/modules/prebid/utils.js]: could this be shortened? since `isInAuRegion()` is true when `!isInUsRegion()` is true? Or does this have a benefit for testing? 🤔 hmmm

```
export const shouldIncludeAppNexus = (): boolean => !isInUsRegion();
```

**Review by katebee — APPROVED** [review]
👍 Grand!

File: static/src/javascripts/projects/commercial/modules/prebid/bid-config.js (hunk 1 above)
Comment (janua) [reply static/src/javascripts/projects/commercial/modules/prebid/bid-config.js]: Yep, Desktop MPU is the default placement ID, which should never happen. Could use the variable but didn't as wanted it to be clear where the default is coming from.

File: static/src/javascripts/projects/commercial/modules/prebid/utils.js [hunk 3]
```diff
@@ -69,7 +70,9 @@ export const getRandomIntInclusive = (
     return Math.floor(Math.random() * (max - min + 1)) + min;
 };
 
-export const shouldIncludeAppNexus = (): boolean => isInAuRegion();
+export const shouldIncludeAppNexus = (): boolean =>
+    isInAuRegion() ||
+    (config.get('switches.prebidAppnexusUkRow') && !isInUsRegion());
```
Comment (janua) [inline static/src/javascripts/projects/commercial/modules/prebid/utils.js]: @katebee Good shout on the logic; I think it's now required because of this switch, but without it it wasn't. What do you think?

**Comment by katebee** [comment]
Does this need to wait on https://trello.com/c/kMkVOr7b ?

Nice advantage of the switch is that this could get merged and then if we do need the keywords done we can delay switching it on 😸

File: static/src/javascripts/projects/commercial/modules/prebid/utils.js [hunk 4]
```diff
@@ -92,6 +92,11 @@ export const shouldIncludeOzone = (): boolean =>
     // include in 1 in 2 (50%) of page views
     !isInUsRegion() && !isInAuRegion() && getRandomIntInclusive(1, 2) === 1;
 
+export const shouldIncludeAppNexus = (): boolean =>
+    isInAuRegion() ||
+    ((config.get('switches.prebidAppnexusUkRow') && !isInUsRegion()) ||
+        !!pbTestNameMap().and);
```
Comment (kelvin-chappell) [inline static/src/javascripts/projects/commercial/modules/prebid/utils.js]: Do you need this `pbTestNameMap`? Looks complicated and there's already a check for `pbtest` in https://github.com/guardian/frontend/blob/884553b99f97ac36ac6463d17d8f78fda5d6a35e/static/src/javascripts/projects/commercial/modules/prebid/bid-config.js#L540

File: static/src/javascripts/lib/url.js [hunk 5]
```diff
@@ -5,6 +5,25 @@ import memoize from 'lodash/functions/memoize';
...
+            .getAll('pbtest')
+            .reduce((acc, value) => {
+                acc[value] = true;
+                return acc;
+            }, {}),
+    (): string =>
+        // Same implicit parameter as the memoized function
+        window.location.search
+);
+
```
Comment (kelvin-chappell) [inline static/src/javascripts/lib/url.js]: Why is this here when it looks completely Prebid-specific?

Comment (janua) [reply static/src/javascripts/lib/url.js]: @kelvin-chappell It made the testing almost impossible; the alternative would be to move it into a commercial url module, what do you think?
@katebee might have thoughts on this too

File: static/src/javascripts/projects/commercial/modules/prebid/utils.js (hunk 4 above)
Comment (janua) [reply static/src/javascripts/projects/commercial/modules/prebid/utils.js]: Initially yes, because we want to test it in these regions with AppNexus UK by sending them a test URL. Once this gets released and OK'd then this can get removed.

**Review by kelvin-chappell — APPROVED** [review]
👍
